cloud/pkg/synccontroller/synccontroller.go

```diff
@@ -222,6 +240,59 @@ func (sctl *SyncController) manageObjectSync(syncs []*v1alpha1.ObjectSync) {
 		}
 	}
 }
+func (sctl *SyncController) DeleteOutdatedObjectSync() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop synccontroller DeleteOutdatedObjectSync loop")
+			return
+		case e := <-sctl.nodeManager.Events():
+			node, ok := e.Object.(*v1.Node)
+			if !ok {
+				klog.Warningf("Object type: %T unsupported", node)
+				continue
+			}
+			switch e.Type {
+			case watch.Deleted:
+				syncs, err := sctl.objectSyncLister.List(labels.Everything())
+				if err != nil {
+					klog.Errorf("Filed to list all the ObjectSyncs: %v", err)
+				}
+				for _, sync := range syncs {
+					nodeName := getNodeName(sync.Name)
+					if isGarbage, err := sctl.objectSyncGarbageCollection(nodeName, sync.Namespace, sync.Name); isGarbage {
```

Comment:
what does the isGarbage  mean here?

Reply:
@fisherxu  isGarbage is true means the object is outdated because edgenode doesn't exist now. we need to delete it.

Reply:
@fisher how about regarding `node deleted event` as a trigger of deleting outdated syncs, not only deleting specific node-related syncs, in consideration of cloudcore crash?

Reply:
@GsssC are you talking to me?